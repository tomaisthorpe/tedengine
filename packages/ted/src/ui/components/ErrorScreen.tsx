import styled from 'styled-components';

const Container = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  background: rgba(150, 51, 51, 1);
  width: 1024px;
  height: 768px;
  font-family: sans-serif;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-transform: uppercase;
  font-size: 14px;
  font-weight: bold;
`;

const Muted = styled.div`
  color: rgba(50, 10, 10, 1);
  font-size: 12px;
  margin-top: 4px;
`;

export default function ErrorScreen({ error }: { error: string }) {
  return (
    <Container>
      <div>It's all gone wrong...try refresh.</div>
      <Muted>{error}</Muted>
    </Container>
  );
}
