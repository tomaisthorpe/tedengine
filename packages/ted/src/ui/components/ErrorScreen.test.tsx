import { render } from '@testing-library/react';
import { ErrorScreen } from './ErrorScreen';

test('renders ErrorScreen component', () => {
  const error = 'Something went wrong';
  const { container } = render(<ErrorScreen error={error} />);
  expect(container).toMatchSnapshot();
});
