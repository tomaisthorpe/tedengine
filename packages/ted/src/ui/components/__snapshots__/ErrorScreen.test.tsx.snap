// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ErrorScreen component 1`] = `
<div>
  <div
    class="sc-bXdtCk hwBjWM"
  >
    <div>
      It's all gone wrong...try refresh.
    </div>
    <div
      class="sc-hKDTPf egyGzG"
    >
      Something went wrong
    </div>
  </div>
</div>
`;
